src/scripts/blocks/logo-slider/DisplayComponent.js

```diff
@@ -0,0 +1,289 @@
+            sizes,
+          } = {},
+        } = {}) => {
+          this.props.setAttributes({
+            slides: [
+              ...this.props.attributes.slides
+                .slice(0, Math.max(0, index)),
+              {
+                ...this.props.attributes.slides[index],
+                imageId,
+                imageUrl,
+                sizes,
+              },
+              ...this.props.attributes
+                .slides.slice(index + 1, this.props.attributes.slides.length),
+            ],
+          });
+        };
+
+  addSlide = () => {
+    const { attributes, setAttributes } = this.props;
+
+    setAttributes({
+      slides: [...attributes.slides, {
+        ...DisplayComponent.emptySlide,
+        id: randId(),
+      }],
+    });
+
+    this.setState({
+      selectedSlide: attributes.slides.length,
+      group: attributes.slides.length + 1,
+    });
+  };
+
+  deleteSlide = (index) => {
+    const { attributes, setAttributes } = this.props;
+
+    if (index === attributes.slides.length - 1) {
+      this.setState({
+        selectedSlide: index - 1,
+        group: (index - 1) > attributes.perSlide ? attributes.slides.length : attributes.perSlide,
+      });
+    }
+
+    this.props.setAttributes({
+      slides: [
+        ...attributes.slides.slice(0, Math.max(0, index)),
+        ...attributes.slides.slice(index + 1, attributes.slides.length),
+      ],
+    });
+  };
+
+  initiateDelete = () => {
+    if (confirm(__('Are you sure you want to delete this logo from the slider?'))) { // eslint-disable-line no-restricted-globals, no-alert
+      this.deleteSlide(this.state.selectedSlide);
+    }
+  };
+
+  selectSlide = index => this.setState({
+    selectedSlide: index,
+  });
+
+  createSelectSlide = index => () => this.selectSlide(index);
+
+  movePrev = () => {
+    const { selectedSlide, group } = this.state;
+
+    const slideOrder = [...this.props.attributes.slides];
+    const temp = slideOrder[selectedSlide];
+    slideOrder[selectedSlide] = slideOrder[selectedSlide - 1];
+    slideOrder[selectedSlide - 1] = temp;
+
+    this.props.setAttributes({
+      slides: slideOrder,
+    });
+
+    this.setState({
+      selectedSlide: selectedSlide - 1,
+      group: (selectedSlide - 1) < this.props.attributes.perSlide ? this.props.attributes.perSlide : group - 1, // eslint-disable-line max-len
+    });
+  };
+
+  moveNext = () => {
+    const { selectedSlide, group } = this.state;
+    const slideOrder = [...this.props.attributes.slides];
+    const temp = slideOrder[selectedSlide];
+    slideOrder[selectedSlide] = slideOrder[selectedSlide + 1];
+    slideOrder[selectedSlide + 1] = temp;
+
+    this.props.setAttributes({
+      slides: slideOrder,
+    });
+
+    this.setState({
+      selectedSlide: selectedSlide + 1,
+      group: (selectedSlide + 1) === (this.props.attributes.slides.length - this.props.attributes.perSlide) ? this.props.attributes.perSlide : group + 1, // eslint-disable-line max-len
```

Comment:
This would be easier to read as a standard conditional statement, rather than a ternary.